File: kernels/Julia/Compositron/examples/dbs_fit.jl
using Printf

using Compositron.Core
using Compositron.DBS


m = Measurement(
    "../../../../testdata/depth-profile_Copper_0000.n42"
)

println(shape(m))

d = m.dbs["OAB"]

default_analyze!(d)

s = d.lineshape_params["S"]
w = d.lineshape_params["W"]
v = d.lineshape_params["V/P"]
p = d.lineshape_params["P/T"]

@printf("  S   = %.6f +/- %.6f\n", s.val, s.err)
@printf("  W   = %.6f +/- %.6f\n", w.val, w.err)
@printf("  V/P = %.6f +/- %.6f\n", v.val, v.err)
@printf("  P/T = %.6f +/- %.6f\n", p.val, p.err)

File: kernels/Rust/compositron/examples/dbs_fit.rs
use compositron::core::Measurement;
use compositron::importers::DataFormat;


fn main() -> anyhow::Result<()> {
    let mut m = Measurement::from_file(
        "../../../../testdata/depth-profile_Copper_0000.n42",
        DataFormat::SlopeN42
    )?;

    let d = m.dbs.get_mut("OAB").unwrap();

    d.default_analyze()?;

    let s = d.lineshape_params.get("S").unwrap();
    let w = d.lineshape_params.get("W").unwrap();
    let v = d.lineshape_params.get("V/P").unwrap();
    let p = d.lineshape_params.get("P/T").unwrap();

    println!("  S   = {:.6} +/- {:.6}", s.val, s.err);
    println!("  W   = {:.6} +/- {:.6}", w.val, w.err);
    println!("  V/P = {:.6} +/- {:.6}", v.val, v.err);
    println!("  P/T = {:.6} +/- {:.6}", p.val, p.err);

    Ok(())
}
